CoastalResilienceNetwork/tnc-network-site
# Implement transition animations from region list view to region details view

## Overview

This PR implements the animated transition from the list view to the details view present in the prototype.

To accomplish this I adapted the code in `main.js` to set the animation: essentially it fades the non-selected list items out according to a timer set by index position, then floats the selected list item to the top of the sidebar before hiding it & replacing with a cloned version in the details view.

We replicate the logic here, adding a deferred that we can resolve once the animation's complete to show the details view. Getting this to work required changing how we instantiate the list view a bit: previously we created it once, then showed or hid it programmatically. Now we create it anew each time the list view's shown and destroy it each time the details view is shown. (We were already doing this create/destroy cycle with the details view.)

Connects #6 

## Demo

Prototype animation:

![tnc-prototype-animation](https://user-images.githubusercontent.com/4165523/32468603-f8fb8cc4-c31c-11e7-8a2d-c1285b25433b.gif)

This PR animation:

![tnc-site-animation](https://user-images.githubusercontent.com/4165523/32468614-01124650-c31d-11e7-948d-9e11d2540acf.gif)

## Testing Instructions
- get this branch, then `server`
- select elements from the list view and verify that you see the animated transition and that the list view's properly re-created on closing the details view
- comment out https://github.com/CoastalResilienceNetwork/tnc-network-site/blob/b3e7f79e332ec6a2468d1d361f6816cf26fb0cbd/src/js/App.js#L78 then reload the app and verify that the list and details views still toggle appropriately but without the animation
- verify that the animation works in IE 11, FF, Chrome, and Safari
